### DISCUSSION TITLE
How to connect local mysql server with Budibase as a "source"?

### DISCUSSION THREAD

**Discussion by lumos-ap** [Budibase/budibase#2520] [Help]
I can not figure out how to add mysql as source. I added the information required but its showing error connecting DB when im trying to fetch tables from the database. 
![image](https://user-images.githubusercontent.com/39933014/131784178-ba17a938-461c-47ec-8ab4-0bbf196129d7.png)

**Comment by shogunpurple** [comment]
Budibase runs in docker, so in order to connect to MySQL running on a local machine your host needs to be:

`host.docker.internal`

You can find more information here:

https://github.com/Budibase/budibase/discussions/1839

**Reply by lumos-ap** [reply]
I tried this but didn't work

**Reply by shogunpurple** [reply]
What operating system are you using?

**Reply by lumos-ap** [reply]
Ubuntu

**Reply by shogunpurple** [reply]
Ah, that's not available on linux. Try `172.17.0.1` - the static IP for the docker host

**Reply by KernOil** [reply]
@lumos-ap  check to see if your MySQL container in docker has a bridged network with the host machine. next, find the IP of that machine. thirdly find the port that the MYSQL container is on it may not be the default 3306.

**Reply by KernOil** [reply]
use "docker ps" to look at your containers and find the port. if the host machine does not have the port forwarding setting set up correctly that can also cause an issue. 
![image](https://user-images.githubusercontent.com/80058238/131869687-7840dda1-4001-4bbc-85e6-bc19fbad374b.png)

**Reply by lumos-ap** [reply]
@shogunpurple That did not work either. Its showing connection time out.

**Reply by lumos-ap** [reply]
@AndrewJRamay Mysql is running on local machine not in any container in docker (if thats what you meant when you asked me to check if my mysl container in docker has a bridged network).

**Reply by KernOil** [reply]
@lumos-ap then use "ifconfig -a" to find your machine's IP and most likely it is the default port 3306. give that a shot
